filament/backend/src/metal/MetalDriver.mm

```diff
@@ -530,8 +530,19 @@
         return;
     }
 
-    mContext->textures.erase(handle_cast<MetalTexture>(th));
-    destruct_handle<MetalTexture>(th);
+    auto* metalTexture = handle_cast<MetalTexture>(th);
+    mContext->textures.erase(metalTexture);
+
+    // Free memory from the texture and mark it as freed.
+    metalTexture->terminate();
+
+    // Delay the destruction of this texture handle.
+    mContext->texturesToDestroy.push_back(th);
+    if (mContext->texturesToDestroy.size() > kMetalFreedTextureListSize) {
+        Handle<HwTexture> handleToFree = mContext->texturesToDestroy.front();
+        mContext->texturesToDestroy.pop_front();
+        destruct_handle<MetalTexture>(handleToFree);
+    }
```

Comment:
the delayed free should be protected by a debug option. Either set at runtime or compile time, but we need a way to turn it off.

Reply:
What about a new option in `DriverConfig`?

```cpp
    struct DriverConfig {
        /*
         * size of handle arena in bytes. Setting to 0 indicates default value is to be used.
         * Driver clamps to valid values.
         */
        size_t handleArenaSize = 0;
        
        size_t textureUseAfterFreePoolSize = 64;
    };

```

Setting to 0 could be the default and disable it.